provider/cmd/crd2pulumi/gen/dotnet.go

```diff
@@ -0,0 +1,203 @@
+		namespaces[groupVersion] = strings.Title(groupPrefix) + "." + versionToUpper(version)
+	}
+	namespaces["meta/v1"] = "Meta.V1"
+
+	// Configure C# language-specific settings. Notice that we set
+	// `compatibility` to `kubernetes20`. This is because the actual ObjectMeta
+	// class defined in the .NET SDK is located at
+	// `Pulumi.Kubernetes.Types.Outputs.Meta.V1.ObjectMeta`. This path would
+	// only get generated properly if `compatibility` was `kubernetes20`.
+	pkg.Language["csharp"] = rawMessage(map[string]interface{}{
+		"compatibility":          "kubernetes20",
+		"dictionaryConstructors": true,
+		"namespaces":             namespaces,
+	})
+
+	files, err := dotnet.GeneratePackage(tool, pkg, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not generate .NET package")
+	}
+
+	files["KubernetesResource.cs"] = []byte(kubernetesResource)
+	files["Utilities.cs"] = []byte(dotNetUtilities)
+
+	// Delete unneeded files
+	for _, unneededFile := range unneededDotNetFiles {
+		delete(files, unneededFile)
+	}
+
+	buffers := map[string]*bytes.Buffer{}
+	for name, code := range files {
+		buffers[name] = bytes.NewBuffer(code)
+	}
+
+	return buffers, nil
+}
+
+// Returns a set of the file paths for each CustomResource class for a generated
+// C# package.
+func (pg *PackageGenerator) dotNetClassPaths() codegen.StringSet {
+	paths := codegen.NewStringSet()
+	for _, crg := range pg.CustomResourceGenerators {
+		for _, groupVersion := range crg.GroupVersions {
+			group, version := splitGroupVersion(groupVersion)
+			path := filepath.Join(strings.Title(groupPrefix(group)), versionToUpper(version), crg.Kind+".cs")
+			paths.Add(path)
+		}
+	}
+	return paths
+}
+
+const kubernetesResource = `// Copyright 2016-2020, Pulumi Corporation
+
+namespace Pulumi.Kubernetes
+{
+    /// <summary>
+    /// A base class for all Kubernetes resources.
+    /// </summary>
+    public abstract class KubernetesResource : CustomResource
+    {
+        /// <summary>
+        /// Standard constructor passing arguments to <see cref="CustomResource"/>.
+        /// </summary>
+        internal KubernetesResource(string type, string name, ResourceArgs? args, CustomResourceOptions? options = null)
+            : base(type, name, args, options)
+        {
+        }
+
+        /// <summary>
+        /// Additional constructor for dynamic arguments received from YAML-based sources.
+        /// </summary>
+        internal KubernetesResource(string type, string name, DictionaryResourceArgs? args, CustomResourceOptions? options = null)
+            : base(type, name, args, options)
+        {
+        }
+    }
+}
+`
+
+// For some reason, we get a `Missing embedded version.txt file` error if we
+// tried running `pulumi up` with the normal `Utilities.cs` file.
+// As a temporary fix, this modified `Utilities.cs` file just removes the
+// `static Utilities()` method.
+const dotNetUtilities = `// *** WARNING: this file was generated by crd2pulumi. ***
```

Comment:
For the time being, here's the `Utilities.cs` we could generate for the crds package, which uses reflection to get the version from the `Pulumi.Kubernetes` assembly:

```c#
// *** WARNING: this file was generated by crd2pulumi. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***

using System;
using System.Reflection;

namespace Pulumi.Kubernetes
{
    static class Utilities
    {
        public static string Version { get; } = GetVersion();

        private static string GetVersion()
        {
            const string UtilitiesType = "Pulumi.Kubernetes.Utilities";
            const string VersionProperty = "Version";

            Type? type = Assembly.GetAssembly(typeof(Provider))?.GetType(UtilitiesType);
            if (type is null)
            {
                throw new InvalidOperationException($"{UtilitiesType} type could not be obtained.");
            }

            PropertyInfo? prop = type.GetProperty(VersionProperty, BindingFlags.Static | BindingFlags.Public);
            if (prop is null)
            {
                throw new InvalidOperationException($"{UtilitiesType}.{VersionProperty} property could not be obtained.");
            }

            var result = prop.GetValue(type, null) as string;
            if (result is null)
            {
                throw new InvalidOperationException($"Expected {UtilitiesType}.{VersionProperty} to return a non-null string.");
            }
            return result;
        }
    }
}
```